app/src/main/java/com/kickstarter/libs/UserCurrency.kt

```diff
@@ -0,0 +1,125 @@
+
+    /**
+     * Returns a currency string appropriate to the user's locale and location relative to a project.
+     *
+     * @param initialValue Value to display, local to the project's currency.
+     * @param project The project to use to look up currency information.
+     */
+    fun format(initialValue: Float, project: Project, symbol: String): String {
+        return format(initialValue, project, false, RoundingMode.DOWN, symbol)
+    }
+
+    /**
+     * Returns a currency string appropriate to the user's locale and location relative to a project.
+     *
+     * @param initialValue Value to display, local to the project's currency.
+     * @param project The project to use to look up currency information.
+     * @param excludeCurrencyCode If true, hide the currency code, even if that makes the returned value ambiguous.
+     * This is used when space is constrained and the currency code can be determined elsewhere.
+     */
+    fun format(initialValue: Float, project: Project,
+               excludeCurrencyCode: Boolean, symbol: String): String {
+
+        return format(initialValue, project, excludeCurrencyCode, RoundingMode.DOWN, symbol)
+    }
+
+    /**
+     * Returns a currency string appropriate to the user's locale and location relative to a project.
+     *
+     * @param initialValue Value to display, local to the project's currency.
+     * @param project The project to use to look up currency information.
+     * @param excludeCurrencyCode If true, hide the currency code, even if that makes the returned value ambiguous.
+     * This is used when space is constrained and the currency code can be determined elsewhere.
+     */
+    fun format(initialValue: Float, project: Project,
+               excludeCurrencyCode: Boolean, roundingMode: RoundingMode, symbol: String): String {
+
+        val currencyOptions = userCurrencyOptions(initialValue, project, symbol)
+
+        val showCurrencyCode = showCurrencyCode(currencyOptions, excludeCurrencyCode)
+
+        val numberOptions = NumberOptions.builder()
+                .currencyCode(if (showCurrencyCode) currencyOptions.currencyCode() else "")
+                .currencySymbol(currencyOptions.currencySymbol())
+                .roundingMode(roundingMode)
+                .build()
+
+        return NumberUtils.format(currencyOptions.value(), numberOptions)
+    }
+
+    /** Show's the project in the user's preferred currency. If the user has no preferred currency the project is shown
+     * in $US as a default.
+     */
+    private fun userCurrencyOptions(value: Float, project: Project, symbol: String): KSCurrency.CurrencyOptions {
+        val fxRate = project.fx_rate()
+        val config = this.currentConfigType.getConfig()
+
+        return if (config.countryCode() == "XX") {
+            KSCurrency.CurrencyOptions.builder()
+                    .country(project.country())
+                    .currencyCode("")
+                    .currencySymbol("US$ ")
```

Comment:
I'm not sure if this true for a US user looking at a US project. I think I'd see the project just as "$"

Reply:
For a regular US user it won't go into that block because it's only for those in un-launched countries.